types/time.go

```diff
@@ -789,30 +789,217 @@ func isValidSeparator(c byte, prevParts int) bool {
 	return prevParts == 2 && (c == ' ' || c == 'T')
 }
 
-// See https://dev.mysql.com/doc/refman/5.7/en/date-and-time-literals.html.
-// The only delimiter recognized between a date and time part and a fractional seconds part is the decimal point.
-func splitDateTime(format string) (seps []string, fracStr string) {
-	index := GetFracIndex(format)
-	if index > 0 {
-		fracStr = format[index+1:]
-		format = format[:index]
+var validIdxCombinations = map[int]struct {
+	h int
+	m int
+}{
+	100: {0, 0}, // 23:59:59Z
+	30:  {2, 0}, // 23:59:59+08
+	50:  {4, 2}, // 23:59:59+0800
+	63:  {5, 2}, // 23:59:59+08:00
+	// postgres supports the following additional syntax that deviates from ISO8601, although we won't support it
+	// currently, it will be fairly easy to add in the current parsing framework
+	// 23:59:59Z+08
+	// 23:59:59Z+08:00
+}
+
+// GetTimezone parses the trailing timezone information of a given time string literal. If idx = -1 is returned, it
+// means timezone information not found, otherwise it indicates the index of the starting index of the timezone
+// information. If the timezone contains sign, hour part and/or minute part, it will be returned as is, otherwise an
+// empty string will be returned.
+//
+// Supported syntax:
+//   MySQL compatible: ((?P<tz_sign>[-+])(?P<tz_hour>[0-9]{2}):(?P<tz_minute>[0-9]{2})){0,1}$, see
+//     https://dev.mysql.com/doc/refman/8.0/en/time-zone-support.html and https://dev.mysql.com/doc/refman/8.0/en/datetime.html
+//     the first link specified that timezone information should be in "[H]H:MM, prefixed with a + or -" while the
+//     second link specified that for string literal, "hour values less than than 10, a leading zero is required.".
+//   ISO-8601: Z|((((?P<tz_sign>[-+])(?P<tz_hour>[0-9]{2})(:(?P<tz_minute>[0-9]{2}){0,1}){0,1})|((?P<tz_minute>[0-9]{2}){0,1}){0,1}))$
+//     see https://www.cl.cam.ac.uk/~mgk25/iso-time.html
+func GetTimezone(lit string) (idx int, tzSign, tzHour, tzSep, tzMinute string) {
```

Comment:
Why not return int64 instead of string?

Reply:
Because the absorb logic bellow, it is better to keep the parsing logic only extract data but not interpret data.